src/main/java/com/google/sps/servlets/GoDataServlet.java

```diff
@@ -84,7 +77,9 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     if (huntEntity == null) {
       response.getWriter().println(ERROR_MSSG);
     } else {
-      String json = (String) huntEntity.getProperty(Constants.HUNT_VAL);
+      String jsonHunt = (String) huntEntity.getProperty(Constants.HUNT_VAL);
+      int curIndex = ((Long) huntEntity.getProperty(Constants.HUNT_INDEX)).intValue();
+      String json = jsonHunt.substring(0, jsonHunt.length() - 1) + ",\"index\":" + curIndex + "}";
```

Comment:
Instead of modifying the string itself (which may be error prone), I think you should construct the JSON using a library of some kind (for example, [JsonObjectBuilder](https://docs.oracle.com/javaee/7/api/javax/json/JsonObjectBuilder.html)).

```
{
  hunt: {
    items: {
    }
  },
  index: -1
}
```
It will add a level of nesting, but also gives you the flexibility to add other fields in the future if you want/need to.